cmd/flow-dps-live/main.go

```diff
@@ -0,0 +1,411 @@
+		log.Error().Str("level", flagLevel).Msg("-a/--all is mutually exclusive with specific indexing flags")
+		pflag.Usage()
+		return failure
+	}
+
+	// Open index database.
+	db, err := badger.Open(dps.DefaultOptions(flagIndex))
+	if err != nil {
+		log.Error().Str("index", flagIndex).Err(err).Msg("could not open index DB")
+		return failure
+	}
+	defer db.Close()
+
+	// Open protocol state database.
+	data, err := badger.Open(dps.DefaultOptions(flagData))
+	if err != nil {
+		log.Error().Err(err).Msg("could not open blockchain database")
+		return failure
+	}
+	defer data.Close()
+
+	// We initialize a metrics logger regardless of whether metrics are enabled;
+	// it will just do nothing if there are no registered metrics.
+	mout := output.New(log, flagMetricsInterval)
+
+	// The storage library is initialized with a codec and provides functions to
+	// interact with a Badger database while encoding and compressing
+	// transparently.
+	var codec dps.Codec
+	codec = zbor.NewCodec()
+	if flagMetrics {
+		size := rcrowley.NewSize("store")
+		mout.Register(size)
+		codec = metrics.NewCodec(codec, size)
+	}
+	storage := storage.New(codec)
+
+	// Check if index already exists.
+	_, err = index.NewReader(db, storage).First()
+	indexExists := err == nil
+	if indexExists && !flagForce {
+		log.Error().Err(err).Msg("index already exists, manually delete it or use (-f, --force) to overwrite it")
+		return failure
+	}
+
+	// The loader component is responsible for loading and decoding the checkpoint.
+	load := loader.New(
+		loader.WithCheckpointPath(flagCheckpoint),
+	)
+
+	nodeID, err := flow.HexStringToIdentifier(flagNodeID)
+	if err != nil {
+		log.Error().Err(err).Msg("invalid node ID")
+		return failure
+	}
+
+	client, err := gcs.NewClient(context.Background())
+	if err != nil {
+		log.Error().Err(err).Msg("could not connect to Google Cloud Platform")
+		return failure
+	}
+
+	bkt := client.Bucket(flagBucket)
+	downloader := bucket.NewDownloader(gcp.NewReader(bkt))
+
+	host, portStr, err := net.SplitHostPort(flagPeerAddr)
+	if err != nil {
+		log.Error().
+			Err(err).
+			Str("peer_address", flagPeerAddr).
+			Msg("invalid peer address format")
+		return failure
+	}
+	port, err := strconv.ParseUint(portStr, 10, 16)
+	if err != nil {
+		log.Error().
+			Err(err).
+			Str("peer_address", flagPeerAddr).
+			Msg("invalid peer port")
+		return failure
+	}
+	key, err := crypto.DecodePublicKeyHex(crypto.ECDSA_P256, flagPeerKey)
+	if err != nil {
+		log.Error().
+			Err(err).
+			Str("peer_key", flagPeerKey).
+			Msg("invalid peer public key")
+		return failure
+	}
+	// FIXME: Get public keys from transit stuff.
+	bootstrapIdentities := []follower.BootstrapNodeInfo{{
+		Host:             host,
+		Port:             uint(port),
+		NetworkPublicKey: key,
+	}}
+	follower, err := follower.NewConsensusFollower(nodeID, bootstrapIdentities, flagBindAddr, follower.WithDataDir(flagData))
```

Comment:
This code is already correct; we just need to make sure that we put the bootstrap information into the data directory's Badger DB the first time before we initialize the consensus follower.